query_resolver.go

```diff
@@ -0,0 +1,103 @@
+package surrealdb
+
+import (
+	"context"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/goccy/go-json"
+)
+
+type QueryResult[T any] struct {
+	Result []T    `json:"result"`
+	Status string `json:"status"`
+	Time   string `json:"time"`
+}
+
+type QueryParams = map[string]any
+
+type QueryResolver[T any] struct {
+	err          error
+	response     *RPCRawResponse
+	queryResults []QueryResult[T]
+	query        string
+	params       QueryParams
+	ctx          context.Context
+}
+
+// Query creates a new query resolver
+// It would be ideal if we create a global for *DB
+func Query[T any](db *DB, ctx context.Context, query string, params ...any) *QueryResolver[T] {
```

Comment:
The context should always be the first parameter here
And creating a global db would be kind of nice to use, but horrendous to test and using multiple db connections would be near impossible